packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -306,6 +308,7 @@ export async function parseImportGlob(
       globs.map((glob) => toAbsoluteGlob(glob, root, importer, resolveId)),
     )
     const isRelative = globs.every((i) => '.!'.includes(i[0]))
+    const onlyKeys = objectKeysRE.test(cleanCode.slice(0, start))
```

Comment:
Doesn't this match any `Object.keys(` that occur before `import.meta.glob`? e.g.

```js
const foo = Object.keys({}) // <-- incorrectly matched
// .. some other code
const result = import.meta.glob('')
```

Reply:
The match here is the string immediately before `import.meta.glob` and ending with `Object.keys(`.

The above code will not match.

Reply:
Ah ok didn't notice the regex had a trailing `$` that will only match the ending part.